webpack.config.js

```diff
@@ -1,28 +1,34 @@
-var path = require("path")
-var webpack = require('webpack')
-var CommonsChunkPlugin = require("webpack/lib/optimize/CommonsChunkPlugin");
+const path = require("path");
+const webpack = require("webpack");
+const CommonsChunkPlugin = require("webpack/lib/optimize/CommonsChunkPlugin");
+const CopyWebpackPlugin = require("copy-webpack-plugin");
 
 module.exports = {
+  context: path.join(__dirname, 'app'),
   entry: {
-    popup: "./app/popup.js",
-    background: "./app/background.js",
-    content_script: "./app/content_script.js"
+    popup: "./popup.tsx",
+    background: "./background.ts",
+    content_script: "./content_script.tsx"
   },
   output: {
     path: path.join(__dirname, 'dist'),
     filename: "[name].js"
   },
+  resolve: {
+    extensions: ["", ".ts", ".tsx", ".js", ".json"]
+  },
   module: {
     loaders: [
       {
-        test: /\.js$/,
-        loaders: ["react-hot", "babel"],
-        include: path.join(__dirname, 'app')
+        test: /\.tsx?$/,
+        loader: "awesome-typescript-loader"
       },
       { test: /\.css$/,
-        loader: "style-loader!css-loader",
-        include: path.join(__dirname, 'app')
+        loader: "style-loader!css-loader"
```

Comment:
nope, I just kept it thinking we might get around to doing it properly next time.